--- btsync.conf	2015-02-01 02:02:59.963918261 -0300
+++ btsync.conf	2015-02-01 02:02:10.917918267 -0300
@@ -4,10 +4,10 @@
 
 /* storage_path dir contains auxilliary app files if no storage_path field: .sync dir created in the directory
    where binary is located. otherwise user-defined directory will be used */
-// "storage_path" : "/home/user/.sync",
+  "storage_path" : "/var/lib/btsync",
 
 /* set location of pid file */
-// "pid_file" : "/var/run/btsync/btsync.pid",
+  "pid_file" : "/var/run/btsync/btsync.pid",
 
 /* use UPnP for port mapping */
   "use_upnp" : true,
