diff -pruN slang-2.1.1.orig/slsh/Makefile.in slang-2.1.1/slsh/Makefile.in
--- slang-2.1.1.orig/slsh/Makefile.in	2007-04-09 07:24:08.000000000 +0200
+++ slang-2.1.1/slsh/Makefile.in	2007-09-22 22:43:46.000000000 +0200
@@ -71,7 +71,7 @@ all: slsh_exe
 slsh_exe: slsh.o readline.o
 	$(CC) $(CFLAGS) slsh.o readline.o -o slsh_exe $(LDFLAGS) $(SRC_LIBS)
 slsh: slsh.o readline.o
-	$(CC) $(CFLAGS) slsh.o readline.o -o slsh $(LDFLAGS) $(INST_LIBS)
+	$(CC) $(CFLAGS) slsh.o readline.o -o slsh $(LDFLAGS) $(SRC_LIBS)
 slsh.o: slsh.c slsh.h config.h Makefile
 	$(CC) -c $(CFLAGS) $(SLANG_SRCINC) $(DEFS) slsh.c
 readline.o: readline.c slsh.h config.h Makefile
